File: app/src/main/java/com/jarvis/bmihealth/presentation/utilx/TypeView.java
package com.jarvis.bmihealth.presentation.utilx;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@IntDef({TypeView.height, TypeView.weight, TypeView.strideLength})
@Retention(RetentionPolicy.SOURCE)
public @interface TypeView {
    int height = 0;
    int weight = 1;
    int strideLength = 2;
}

File: app/src/main/java/com/jarvis/bmihealth/common/enums/TypeView.kt
package com.jarvis.bmihealth.common.enums

//@IntDef(TypeView.height, TypeView.weight, TypeView.strideLength)
//@Retention(RetentionPolicy.SOURCE)
//annotation class TypeView {
//    companion object {
//        var height = 0
//        var weight = 1
//        var strideLength = 2
//    }
//}

enum class TypeView(val index: Int) {
    HEIGHT(0),
    WEIGHT(1),
    STRIDE_LENGTH(2)
}

